core/domain/draft_upgrade_services.py

```diff
@@ -207,6 +207,48 @@ def _convert_html_in_draft_change_list(
                 })
         return draft_change_list
 
+    @classmethod
+    def _convert_states_v51_dict_to_v52_dict(cls, draft_change_list):
+        """Converts from version 51 to 52. Version 52 fixes content IDs
+        in translations and voiceovers. As this only affects content IDs and
+        therefore doesn't affect any pre-existing drafts, there should be
```

Comment:
I don't follow this -- why does it not affect pre-existing drafts? I can still make an edit through the editor translation tab and save it, right?

Also there seems to be nontrivial logic in the implementation of this function which I think contradicts this docstring.

Reply:
Updated.